### PR TITLE
Enable avb_streamhandler_demo in make install

### PR DESCRIPTION
**Pull request by keerockl**
1. add below lib and exe for make install
 /usr/lib64/libias-media_transport-avb_streamhandler.so
 /usr/lib64/pluginias-media_transport-avb_configuration_reference.so
 /usr/bin/avb_streamhandler_demo
2. setcap to /usr/bin/avb_streamhandler_demo
3. modify setcap.sh

### CODE REVIEW

File: CMakeLists.avb_streamhandler.txt [hunk 1]
```diff
@@ -89,3 +89,8 @@ set_target_properties( ias-media_transport-avb_streamhandler PROPERTIES VERSION
 target_link_libraries( ias-media_transport-avb_streamhandler ias-audio-common )
 target_link_libraries( ias-media_transport-avb_streamhandler ias-media_transport-lib_ptp_daemon )
 target_link_libraries( ias-media_transport-avb_streamhandler ias-media_transport-avb_helper )
+if( "x86_64" STREQUAL ${CMAKE_SYSTEM_PROCESSOR})
+    install(TARGETS ias-media_transport-avb_streamhandler DESTINATION lib64)
+else()
+    install(TARGETS ias-media_transport-avb_streamhandler DESTINATION lib)
+endif()
```
Comment (edersondisouza) [inline CMakeLists.avb_streamhandler.txt]: Ha, this wouldn't work on ArchLinux, where lib is a symlink to lib64, and lib32 is different. Isn't cmake itself able to figure this out? See https://cmake.org/cmake/help/v3.0/module/GNUInstallDirs.html

File: setcap.sh [hunk 2]
```diff
@@ -11,14 +11,11 @@
...
 	$AVB_SETCAP_TOOL $1 && exit
-	echo "Failed"
-fi
-
-which setcap_tool &> /dev/null
-if [ $? -eq 0 ]; then
-	echo "Trying to use setcap_tool on PATH env var"
-	setcap_tool $1 && exit
-	echo "Failed"
+	echo "Failed to set cap"
```
Comment (edersondisouza) [inline setcap.sh:14]: First `if` above also deals with `setcap_tool`. Here, I would just add one more check, after this: if no `setcap_tool` was found, then we fallback to `setcap` We can create an issue so we can do a proper remove of `setcap_tool` later.

**Review by edersondisouza — CHANGES_REQUESTED** [review]

File: CMakeLists.avb_streamhandler.txt (hunk 1 above)
Comment (keerockl) [reply CMakeLists.avb_streamhandler.txt]: Clearlinux returns wrong values.

CMAKE_INSTALL_LIBDIR: /usr/lib64, 
CMAKE_INSTALL_FULL_LIBDIR: 
CMAKE_INSTALL_BINDIR:

Comment (keerockl) [reply CMakeLists.avb_streamhandler.txt]: I have reported to Clearlinux team. for now I will be using lib64 only until it gets fixed from Clearlinux.

File: setcap.sh (hunk 2 above)
Comment (keerockl) [reply setcap.sh:14]: I will add back the case.

File: CMakeLists.avb_streamhandler.txt (hunk 1 above)
Comment (edersondisouza) [reply CMakeLists.avb_streamhandler.txt]: At least the most difficult one, LIBDIR, appears correct. What else is necessary?

Comment (keerockl) [reply CMakeLists.avb_streamhandler.txt]: no, it is supposed to return as just lib64 without /usr/lib64.
another option would be to provide command prompt paramater such as -DLIB_DIR=lib64 or lib when running make install. what do you think?

Comment (edersondisouza) [reply CMakeLists.avb_streamhandler.txt]: First sanity check is, did you remember to add the GNUInstallDirs module? With `include(GNUInstallDirs)`?
Second thing, how did you check the variable values? I've done some quick tests here and install target was quite sane with them.
Finally,  I'd make something like
`install(TARGETS ias-media_transport-avb_streamhandler DESTINATION ${CMAKE_INSTALL_LIBDIR})` and if any system needs a different value for CMAKE_INSTALL_LIBDIR, simply instruct people/packagers to use `cmake (...) -DCMAKE_INSTALL_LIBDIR=<correct-path>`.

Comment (keerockl) [reply CMakeLists.avb_streamhandler.txt]: Ah, I didn't include module. 
it returns correct values now.

CMAKE_INSTALL_LIBDIR: lib64, CMAKE_INSTALL_FULL_LIBDIR:/usr/local/lib64, CMAKE_INSTALL_BINDIR:bin

you can printout value by below line in cmake.
message( "CMAKE_INSTALL_LIBDIR: ${CMAKE_INSTALL_LIBDIR}, CMAKE_INSTALL_FULL_LIBDIR:${CMAKE_INSTALL_FULL_LIBDIR}, CMAKE_INSTALL_BINDIR:${CMAKE_INSTALL_BINDIR}" )

Comment (edersondisouza) [reply CMakeLists.avb_streamhandler.txt]: Ohh... nice to know =D

Comment (keerockl) [reply CMakeLists.avb_streamhandler.txt]: One more issue I have is I am able to install them into /usr/local/lib64 however system by default is not set to find lib in /usr/local/lib64 folder. instead it only looks for /usr/lib64.

Comment (edersondisouza) [reply CMakeLists.avb_streamhandler.txt]: I don't think that's an issue. For those systems, simply set "-DCMAKE_INSTALL_PREFIX=/usr" when calling cmake.

Comment (keerockl) [reply CMakeLists.avb_streamhandler.txt]: sounds good. please review my second patch then.

Comment (keerockl) [reply CMakeLists.avb_streamhandler.txt]: cmake for make install in clearlinux

env PKG_CONFIG_PATH=$AVB_DEPS/lib/pkgconfig cmake -DIAS_IS_HOST_BUILD=1 -DIAS_DISABLE_DOC=1 -DCMAKE_INSTALL_PREFIX=/usr ../

File: CMakeLists.avb_streamhandler_demo.txt [hunk 3]
```diff
@@ -29,7 +29,7 @@ if (${IAS_IS_HOST_BUILD})
   )
 endif()
 
-install(TARGETS avb_streamhandler_demo DESTINATION bin)
+install(TARGETS avb_streamhandler_demo DESTINATION CMAKE_INSTALL_BINDIR)
```
Comment (edersondisouza) [inline CMakeLists.avb_streamhandler_demo.txt]: Shouldn't be `${CMAKE_INSTALL_BINDIR}`, like in the others?

File: setcap.sh [hunk 4]
```diff
@@ -12,7 +12,13 @@ if [ -x "$AVB_SETCAP_TOOL" ]; then
 	echo "Using setcap_tool defined on AVB_SETCAP_TOOL env var"
 	$AVB_SETCAP_TOOL $1 && exit
 	echo "Failed to set cap"
-elif [ `which setcap` ]; then
+fi
+if [ `which setcap_tool &> /dev/null ` ]; then
```
Comment (edersondisouza) [inline setcap.sh]: I'm not sure if it's a good idea to change the `elif` for `if`. Supposed that first setcap_tool fails because current user doesn't have permission to do so. It will fail again up to more two times when trying again.

Comment (keerockl) [reply setcap.sh]: we will keep as it is for not and will re-visit later time when this becomes issue.

**Comment by edersondisouza** [comment]
Looks good. I think that you can squash those patches before merging, right?

File: setcap.sh [hunk 5]
```diff
@@ -11,14 +11,17 @@
...
-
-which setcap_tool &> /dev/null
-if [ $? -eq 0 ]; then
+if [ `which setcap_tool &> /dev/null ` ]; then
 	echo "Trying to use setcap_tool on PATH env var"
 	setcap_tool $1 && exit
-	echo "Failed"
+	echo "Failed to set cap"
+fi
+if [ `which setcap` ]; then
```
Comment (edersondisouza) [inline setcap.sh]: Oh, before you merge, could you add `&> /dev/null` here, so that it doesn't pollute output?

**Review by edersondisouza — APPROVED** [review]

File: setcap.sh [hunk 6]
```diff
@@ -11,14 +11,19 @@
 if [ -x "$AVB_SETCAP_TOOL" ]; then
 	echo "Using setcap_tool defined on AVB_SETCAP_TOOL env var"
 	$AVB_SETCAP_TOOL $1 && exit
-	echo "Failed"
+	echo "Failed to set cap"
 fi
-
 which setcap_tool &> /dev/null
-if [ $? -eq 0 ]; then
+if [ $? == 0 ]; then
```
Comment (edersondisouza) [inline setcap.sh]: Isn't `==` a bashism that is not portable? Why did you change the `-eq`?

**Review by edersondisouza — APPROVED** [review]

**Comment by keerockl** [comment]
pull request is merged. https://github.com/intel/AVBStreamHandler/commit/f7054bebc26c5892fab21699d6812520fae0c73d

Please use following command for make install in Clearlinux
env PKG_CONFIG_PATH=$AVB_DEPS/lib/pkgconfig cmake -DIAS_IS_HOST_BUILD=1 -DIAS_DISABLE_DOC=1 -DCMAKE_INSTALL_PREFIX=/usr ../

or if using default path for package config
cmake -DIAS_IS_HOST_BUILD=1 -DIAS_DISABLE_DOC=1 -DCMAKE_INSTALL_PREFIX=/usr ../
